pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -561,3 +564,34 @@ func (c *Cluster) resolveNode(nodeName string) *rookalpha.Node {
 	rookNode.Resources = k8sutil.MergeResourceRequirements(rookNode.Resources, c.resources)
 	return rookNode
 }
+
+func (c *Cluster) getValidStorageNodes() []rookalpha.Node {
+	storageNodes := c.Storage.Nodes
+	nodeOptions := metav1.ListOptions{}
+	nodeOptions.TypeMeta.Kind = "Node"
+	allNodes, err := c.context.Clientset.CoreV1().Nodes().List(nodeOptions)
+	if err != nil {
+		// cannot list nodes, assume the specified nodes are valid
+		logger.Warningf("failed to list nodes: %v", err)
+		return storageNodes
```

Comment:
seems like we should fail the osd orchestration if we can't list the nodes. The storageNodes list is the desired state, but might not be actual nodes in the cluster, which would lead to other errors.

Reply:
done